.gitignore

```diff
@@ -0,0 +1,96 @@
+
+# Created by https://www.gitignore.io/api/java,intellij
+
+### Intellij ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and Webstorm
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff:
+.idea/*
+.classpath
+.project
+.settings/*
+target/*
```

Comment:
FYI: it's possible to include maven in www.gitignore.io and it adds `target` automatically.